src/link/Wasm.zig

```diff
@@ -680,6 +680,11 @@ fn linkWithLLD(self: *Wasm, comp: *Compilation) !void {
             // Put stack before globals so that stack overflow results in segfault immediately
             // before corrupting globals. See https://github.com/ziglang/zig/issues/4496
             try argv.append("--stack-first");
+
+            // Reactor execution model does not have _start so lld doesn't look for it.
+            if (self.base.options.wasi_exec_model != null and self.base.options.wasi_exec_model.? == Compilation.WasiExecModel.reactor) {
+                try argv.append("--no-entry");
```

Comment:
I think it is more readable to unpack the optional using the `if (opt) |value|` syntax which is more readable, and use labeled block to leave early if the inner value is not a `.reactor`:

```zig
if (self.base.options.wasi_exec_model) |exec_model| blk: {
    if (exec_model != .reactor) break :blk;
    try argv.append("--no-entry");
}
```

See [Optionals](https://ziglang.org/documentation/0.8.0/#toc-Optionals) for more details.